lib/src/main/java/com/knuddels/jtokkit/LazyEncodingRegistry.java

```diff
@@ -0,0 +1,26 @@
+package com.knuddels.jtokkit;
+
+import com.knuddels.jtokkit.api.Encoding;
+import com.knuddels.jtokkit.api.EncodingRegistry;
+import com.knuddels.jtokkit.api.EncodingType;
+
+import java.util.Optional;
+
+/**
+ * A lazy initialization implementation of {@link EncodingRegistry}. It does not register any encoding until either the 
+ * {@link #getEncoding(EncodingType)} or {@link #getEncoding(String)} method is called. 
+ * When one of these methods is called, the requested {@link EncodingType} is registered.
+ */
+public class LazyEncodingRegistry extends AbstractEncodingRegistry {
+    @Override
+    public Encoding getEncoding(EncodingType encodingType) {
+        addEncoding(encodingType);
+        return super.getEncoding(encodingType);
+    }
+
+    @Override
+    public Optional<Encoding> getEncoding(String encodingName) {
+        addEncoding(EncodingType.valueOf(encodingName));
```

Comment:
The getEncoding(String) method is the only method that registered custom encodings can be retrieved. In this current implementation, it will throw, if encodingName does not correspond to a pre-defined encoding. This leads to the LazyEncodingRegistry being unusable with custom encodings, which is not what we want :smile: 

The flow of this method should be:
* Check if a encoding for that string is registered (if yes, return it)
* If not, check if the encodingName matches a pre-defined encoding
* If yes, re-use the lazy initialization flow of the getEncoding(EncodingType) method
* It not, return an empty optional